numba/tests/test_linalg.py

```diff
@@ -236,5 +244,72 @@ def test_matmul_mm(self):
         self.check_dot_mm(matmul_usecase, None, "'@'")
 
 
+def invert_matrix(a):
+    return np.linalg.inv(a)
+
+
+class TestLinalgInv(TestCase):
+    """
+    Tests for np.linalg.inv.
+    """
+
+    dtypes = (np.float64, np.float32, np.complex128, np.complex64)
+
+    def sample_vector(self, n, dtype):
+        # Be careful to generate only exactly representable float values,
+        # to avoid rounding discrepancies between Numpy and Numba
+        base = np.arange(n)
+        if issubclass(dtype, np.complexfloating):
+            return (base * (1 - 0.5j) + 2j).astype(dtype)
+        else:
+            return (base * 0.5 + 1).astype(dtype)
+
+    def sample_matrix(self, m, dtype, order):
+        a = np.zeros((m, m), dtype, order)
+        a += np.diag(self.sample_vector(m, dtype))
+        return a
+
+    def assert_value_error(self, cfunc, args, msg):
+        with self.assertRaises(ValueError) as raises:
+            cfunc(*args)
+        self.assertIn(msg, str(raises.exception))
+
+    def assert_non_square(self, cfunc, args):
+        msg = "np.linalg.inv can only work on square arrays."
+        self.assert_value_error(cfunc, args, msg)
+
+    def assert_wrong_dtype(self, cfunc, args):
+        pass
+
+    def assert_singular_matrix(self, cfunc, args):
+        msg = "Matrix is singular and cannot be inverted"
+        self.assert_value_error(cfunc, args, msg)
+
+    @needs_lapack
+    def test_linalg_inv(self):
+        """
+        Test np.linalg.inv
+        """
+        n = 10
+        cfunc = jit(nopython=True)(invert_matrix)
+        for dtype, order in product(self.dtypes, 'CF'):
+            a = self.sample_matrix(n, dtype, order)
```

Comment:
Please could you add a test with a non-trivial matrix (just to make sure things which could get in a mess with respect to transposes and `CF` ordering are ok)?